Source/controls/plrctrls.cpp

```diff
@@ -1508,16 +1508,21 @@ void plrctrls_after_game_logic()
 void UseBeltItem(int type)
 {
 	for (int i = 0; i < MAXBELTITEMS; i++) {
-		auto &myPlayer = Players[MyPlayerId];
-		const int id = AllItemsList[myPlayer.SpdList[i].IDidx].iMiscId;
-		const int spellId = AllItemsList[myPlayer.SpdList[i].IDidx].iSpell;
-		if ((type == BLT_HEALING && (id == IMISC_HEAL || id == IMISC_FULLHEAL || (id == IMISC_SCROLL && spellId == SPL_HEAL)))
-		    || (type == BLT_MANA && (id == IMISC_MANA || id == IMISC_FULLMANA))
-		    || id == IMISC_REJUV || id == IMISC_FULLREJUV) {
-			if (!myPlayer.SpdList[i].isEmpty()) {
-				UseInvItem(MyPlayerId, INVITEM_BELT_FIRST + i);
-				break;
-			}
+		Item &item = Players[MyPlayerId].SpdList[i];
+		if (item.isEmpty()) {
+			continue;
+		}
+
+		const item_misc_id id = AllItemsList[item.IDidx].iMiscId;
+		const spell_id spellId = AllItemsList[item.IDidx].iSpell;
+
+		bool isRejuvenation = IsAnyOf(id, IMISC_REJUV, IMISC_FULLREJUV);
+		bool isHealing = isRejuvenation || IsAnyOf(id, IMISC_HEAL, IMISC_FULLHEAL) || (id == IMISC_SCROLL && spellId == SPL_HEAL);

```

Comment:
If we use `item` directly, we could use some helper functions of it:
```suggestion
		bool isHealing = isRejuvenation || IsAnyOf(id, IMISC_HEAL, IMISC_FULLHEAL) || item.IsScrollOf(SPL_HEAL);
```

Reply:
`IsScrollOf` also includes targeted scrolls, assuming that they all use the correct spell id of `SPL_HEALOTHER` though so this should be functionally the same.

Reply:
Applied